src/core/column/mean.h

```diff
@@ -25,20 +25,20 @@
 namespace dt {
 
 
-template <typename T, bool IS_GROUPED>
-class Mean_ColumnImpl : public ReduceUnary_ColumnImpl<T, IS_GROUPED> {
+template <typename T, typename U, bool IS_GROUPED>
```

Comment:
We don't really need to pass `U` for mean, minmax and sumprod calculations, because it is identical to `T`.